data_utils.py

```diff
@@ -63,118 +62,76 @@ def get_dataset_loader(config, dataset, shuffle=False, train=True):
+    val_path = config[f'val_path'] or os.path.join(config.data_dir, 'valid.txt')
+    if os.path.exists(val_path):
+        datasets['val'] = _load_raw_data(val_path)
+    else:
+        datasets['train'], datasets['val'] = train_test_split(
+            datasets['train'], test_size=config.val_size, random_state=42)
 
-    log.info(f"Finish loading dataset (train: {len(datasets['train'])} / test: {len(datasets['test'])} / dev: {len(datasets['dev'])})")
+    msg = ' / '.join(f'{k}: {len(v)}' for k, v in datasets.items())
+    log.info(f'Finish loading dataset ({msg})')
     return datasets
 
 
-def _get_tokenizer():
-    def caml_tokenizer(text):
-        tokenizer = RegexpTokenizer(r'\w+')
-        return [t.lower() for t in tokenizer.tokenize(text) if not t.isnumeric()]
-
-    # attention xml
-    # [token.lower() if token != sep else token for token in word_tokenize(sentence)
-    #     if len(re.sub(r'[^\w]', '', token)) > 0]
-    return caml_tokenizer
-
-
-def _preprocess_on_split(dataset):
-    # split text: https://pytorch.org/text/_modules/torchtext/data/utils.html
-    # tokenizer = get_tokenizer(tokenizer=None)
-    tokenizer = _get_tokenizer()
-    dataset['text'] = [tokenizer(text) for text in dataset['text']]
-    dataset = [dict(zip(dataset, v)) for v in zip(*dataset.values())]
-    dataset = [d for d in dataset if len(d['label']) > 0]
-
-    return dataset
-
-
-def build_text_dict(examples, min_vocab_freq, vocab_file=None):
-    if vocab_file:
-        log.info(f'Load vocab from {vocab_file}')
-        with open(vocab_file, 'r') as f:
-            vocab_list = ['**PAD**'] + [vocab.strip() for vocab in f.readlines()]
-        vocabs = Vocab(collections.Counter(vocab_list), specials=['<unk>'], min_freq=1, specials_first=False)
+def load_or_build_text_dict(config, dataset):
+    if config.vocab_file:
+        log.info(f'Load vocab from {config.vocab_file}')
+        with open(config.vocab_file, 'r') as fp:
+            vocab_list = ['**PAD**'] + [vocab.strip() for vocab in fp.readlines()]
+        vocabs = Vocab(collections.Counter(vocab_list), specials=['<unk>'],
+                       min_freq=1, specials_first=False)
     else:
         counter = collections.Counter()
-        for example in examples:
-            unique_tokens = set(example['text'])
-            counter.update(unique_tokens)
-        vocabs = Vocab(counter, specials=['<pad>', '<unk>'], min_freq=min_vocab_freq)
-
+        for data in dataset:
+            counter.update(set(data['text']))
+        vocabs = Vocab(counter, specials=['<pad>', '<unk>'],
+                       min_freq=config.min_vocab_freq)
     log.info(f'Read {len(vocabs)} vocabularies.')
     return vocabs
 
 
+def load_or_build_label(config, datasets):
+    if config.label_file:
+        log.info('Load labels from {config.label_file}')
+        with open(config.label_file, 'r') as fp:
+            classes = sorted([s.strip() for s in fp.readlines()])
```

Comment:
Now we assume labels are in the data files.. So do we still need this if config.label_file?

Reply:
It's for the zero-shot setting in the future.